src/expr-parser/src/parser.rs

```diff
@@ -0,0 +1,1483 @@
+                } else if input.eat(syn::Token![*]) {
+                    exp_opd = true;
+                    Op::Bin(mz_expr::BinaryFunc::MulInt64) // TODO: fix placeholder
+                } else if input.eat(syn::Token![/]) {
+                    exp_opd = true;
+                    Op::Bin(mz_expr::BinaryFunc::DivInt64) // TODO: fix placeholder
+                } else if input.eat(syn::Token![%]) {
+                    exp_opd = true;
+                    Op::Bin(mz_expr::BinaryFunc::ModInt64) // TODO: fix placeholder
+                } else if input.eat(kw::AND) {
+                    exp_opd = true;
+                    Op::Var(mz_expr::VariadicFunc::And)
+                } else if input.eat(kw::OR) {
+                    exp_opd = true;
+                    Op::Var(mz_expr::VariadicFunc::Or)
+                } else if input.eat(kw::IS) {
+                    let negate = input.eat(kw::NOT);
+
+                    let lookahead = input.lookahead1();
+                    let func = if input.look_and_eat(kw::NULL, &lookahead) {
+                        mz_expr::func::IsNull.into()
+                    } else if input.look_and_eat(kw::TRUE, &lookahead) {
+                        mz_expr::func::IsTrue.into()
+                    } else if input.look_and_eat(kw::FALSE, &lookahead) {
+                        mz_expr::func::IsFalse.into()
+                    } else {
+                        Err(lookahead.error())?
+                    };
+
+                    if negate {
+                        Op::Neg(func)
+                    } else {
+                        Op::Unr(func)
+                    }
+                } else {
+                    // We were expecting an optional operator but didn't anything.
+                    // Exit the parsing loop and process the postfix vector.
+                    break;
+                };
+
+                // First, pop the operators which are already on the opstack that
+                // have higher or equal precedence than the current operator and
+                // append them to the postfix.
+                while opstack
+                    .last()
+                    .map(|op1: &Op| op1.precedence() >= op.precedence())
+                    .unwrap_or(false)
+                {
+                    let op1 = opstack.pop().expect("non-empty opstack");
+                    postfix.push(Entry::Operator(op1));
+                }
+
+                // Then push the op from this iteration onto the stack.
+                opstack.push(op);
+            }
+        }
+
+        // Pop all remaining symbols from opstack and append them to postfix.
+        postfix.extend(opstack.into_iter().rev().map(Entry::Operator));
+
+        if postfix.is_empty() {
+            let msg = "Cannot parse an empty expression";
+            Err(Error::new(input.span(), msg))?
+        }
+
+        // Flatten the postfix vector into a single MirScalarExpr.
+        let mut stack = vec![];
+        postfix.reverse();
```

Comment:
`postfix` is actually more like `prefix` after this `reverse()`, if I'm reading the code correctly. Maybe create a new variable. Or just traverse `postfix` with a for loop instead of a while loop that pops.

Reply:
Reversing the `postfix` notation does not automatically give you a `prefix` notation, as demonstrated by the following counter-example:

* postfix : `AB+CD-*`
* reversed postfix : `*-DC+BA`
* prefix :  `*+AB-CD`

so using `prefix` might be confusing (but `postfix_rev` sounds reasonable).

The reason I am doing the `postfix.reverse()` here is that I want to process the elements front-to-back in the loop below, where `.pop()` is constant time but taking an element from the front is not. I've now rewritten this as 
```rust
for entry in postfix.into_iter().rev() {
    // ...
}
```
without the reversal, hopefully this is less confusing.